src/CaBundle.php

```diff
@@ -130,21 +128,17 @@ public static function getSystemCaRootBundlePath(LoggerInterface $logger = null)
      */
     public static function getBundledCaBundlePath()
     {
-        $caBundleFile = __DIR__.'/../res/cacert.pem';
+        $caBundleFile = __DIR__. str_replace('/', DIRECTORY_SEPARATOR, '/../res/cacert.pem');
 
         // cURL does not understand 'phar://' paths
         // see https://github.com/composer/ca-bundle/issues/10
         if (0 === strpos($caBundleFile, 'phar://')) {
-            file_put_contents(
-                $tempCaBundleFile = tempnam(sys_get_temp_dir(), 'openssl-ca-bundle-'),
-                file_get_contents($caBundleFile)
-            );
-
-            register_shutdown_function(function() use ($tempCaBundleFile) {
-                @unlink($tempCaBundleFile);
-            });
+            $file = tmpfile();
+            $filePath = stream_get_meta_data($file);
+            $filePath = $filePath['uri'];
+            copy($caBundleFile, $filePath);
 
-            $caBundleFile = $tempCaBundleFile;
+            $caBundleFile = $filePath;
```

Comment:
The changes here I also don't really see what the point is except change for the sake of change.. Which I'd rather avoid as it's not broken right now.

Reply:
Was more to remove the register shutdown function. Can remove it though

Reply:
I see, the comments of tmpfile specifically list the code we had as a workaround for "some environments" though, so I'd rather keep it as is and make sure it works everywhere as this lib's main goal is to ensure we get a working CA bundle under all conditions.

Reply:
Reverted.